demo/laneLoader.js

```diff
@@ -481,6 +481,91 @@ function addInvalidAnnotations (laneGeometries) {
   }
 }
 
+// Adds invalid annotations split into multiple layers
+function addInvalidAnnotationsSplit (laneGeometries, invalidLayerSize) {
+  const { leftInvalidAnnotations, rightInvalidAnnotations } = laneGeometries;
+  const aRoot = viewer.scene.annotations;
+
+  const aInvalid = new Potree.Annotation({
+    title: 'Invalid Lanes',
+    position: null,
+    collapseThreshold: 0
+  });
+  aRoot.add(aInvalid);
+
+  if (leftInvalidAnnotations.length !== 0) {
+    aInvalid.position = leftInvalidAnnotations[0].position;
+    const leftLayer = new Potree.Annotation({
+      title: "Left",
+      position: null,
+      collapseThreshold: 0
+    });
+    aInvalid.add(leftLayer);
+
+    for (let i = 0, len = leftInvalidAnnotations.length; i < len; i += invalidLayerSize) {
+      aInvalid.position = leftInvalidAnnotations[i].position;
+      leftLayer.position = leftInvalidAnnotations[i].position;
+      const aLeft = new Potree.Annotation({
+        title: `${i + 1} - ${Math.min(i + invalidLayerSize, leftInvalidAnnotations.length)}`,
+        position: leftInvalidAnnotations[i].position,
+        collapseThreshold: 0
+      });
+      aLeft.visible = false;
+      leftLayer.children.push(aLeft);
+      aLeft.parent = leftLayer;
+
+      const invalidArray = [];
+      for (let ii = i, len = Math.min(i + invalidLayerSize, leftInvalidAnnotations.length); ii < len; ii++) {
+        const { tag, position } = leftInvalidAnnotations[ii];
+        const invalidAnnotation = new Potree.Annotation({
+          title: tag,
+          position,
+          cameraPosition: new THREE.Vector3(position.x, position.y, position.z + 20),
+          cameraTarget: position
+        });
+        invalidArray.push(invalidAnnotation);
+      }
+      aLeft.addMultiple(viewer.scene.annotations, invalidArray);
+    }
```

Comment:
I don't like the `leftLayer.children.push(aLeft);`. Why not collect all the children via `leftLayer.children = leftInvalidAnnotations .map(...)`?

And building the `invalidArray` by `.push()` will lead to unnecessary memory fragmentation. It will be mildly awkward, but you could make the entire `invalidArray = new Array(len);` at once, and then fill in the elements.